src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        User user = optionalUser.orElseThrow(
+                () -> new UsernameNotFoundException("Can't find user with email: " + email));
+        org.springframework.security.core.userdetails.User.UserBuilder userBuilder =
```

Comment:
from checklist:
When you create UserBuilder, move package name from method to imports.